zp-core/zp-extensions/downloadList.php

```diff
@@ -647,25 +699,8 @@ function printDownloadAlbumZipURL($linktext = NULL, $albumobj = NULL, $fromcache
 			exitZP(); //	terminate the script with no output
 		}
 	}
-	$hash = getOption('downloadList_password');
-	if (GALLERY_SECURITY != 'public' || $hash) {
-		//	credentials required to download
-		if (!zp_loggedin((getOption('downloadList_rights')) ? FILES_RIGHTS : ALL_RIGHTS)) {
-			$user = getOption('downloadList_user');
-			zp_handle_password('zpcms_auth_download', $hash, $user);
-			if ((!empty($hash) && zp_getCookie('zpcms_auth_download') != $hash)) {
-				$show = ($user) ? true : NULL;
-				$hint = '';
-				if (!empty($hash)) {
-					$hint = get_language_string(getOption('downloadList_hint'));
-				}
-				if (isset($_GET['albumzip'])) {
-					$item .= '&albumzip';
-				}
-				printPasswordForm($hint, true, $show, '?download=' . $item);
-				exitZP();
-			}
-		}
+	if (!DownloadList::checkAccess($hint, $show)) {
+		return;
```

Comment:
If check access fails return to zp-core/index.php, that is now ready to show the password page.